csharp/ServiceAccountMaintenance/Program.cs

```diff
@@ -0,0 +1,145 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Google.Protobuf.Collections;
+using Google.Protobuf.WellKnownTypes;
+using Improbable.SpatialOS.ServiceAccount.V1Alpha1;
+
+namespace ServiceAccountMaintenance
+{
+    internal class Program
+    {
+        /// <summary>
+        ///     PLEASE REPLACE ME.
+        ///     Your SpatialOS project name.
+        ///     This scenario will create a new service account with read and write access to this project.
+        /// </summary>
+        private const string ProjectName = "platform_sdk_examples";
+
+        /// <summary>
+        ///     The threshold for when a service account's expiry time should be increased, in days relative to
+        ///     the current time. If the service account expires in fewer days than this, its lifetime will be extended
+        ///     by DaysToExpandServiceAccountBy days.
+        /// </summary>
+        private const int DaysRemainingAtWhichExpiryShouldBeIncreased = 1;
+
+        /// <summary>
+        ///     How many days to expand the service account's lifetime by if it is too close to expiry (as defined by
+        ///     DaysRemainingAtWhichExpiryShouldBeIncreased), relative to the current time.
+        /// </summary>
+        private const int DaysToExpandServiceAccountBy = 4;
+
+        private const int NumberOfServiceAccountsToCreate = 10;
+
+        /// <summary>
+        ///     This list keeps track of the IDs of the service accounts which you create during setup.
+        /// </summary>
+        private static List<long> ServiceAccountIds;
+
+        /// <summary>
+        ///     PLEASE REPLACE ME.
+        ///     The name given to service accounts created during setup.
+        /// </summary>
+        private const string ServiceAccountName = "sa_maintenance_scenario";
+
+        private static readonly ServiceAccountServiceClient ServiceAccountServiceClient =
```

Comment:
Could we pls allow injecting refresh token through environment variables like in the other examples? This would make it easier for us to run the scenario in CI.